dashboard/src/client/client.ts

```diff
@@ -188,7 +186,24 @@ export class MatterClient {
     if (event.event === "node_removed") {
       delete this.nodes[event.data];
       this.nodes = { ...this.nodes };
-      console.log("node removed!")
+      this.fireEvent("nodes_changed");
+      return;
+    }
+
+    if (event.event === "node_updated") {
+      const node = new MatterNode(event.data);
+      this.nodes = { ...this.nodes, [node.node_id]: node };
+      this.fireEvent("nodes_changed");
+      return;
+    }
+
+    if (event.event === "attribute_updated") {
+      const nodeId = (event.data as Array<any>)[0] as number;
```

Comment:
Why isn't `data` typed as array for the `ServerEventAttributeUpdated` event? It is now typed as an object?

Reply:
because it can contain anything, it is literally any

Reply:
ah but wait... for the attribute_updated_event its actually always a tuple

Reply:
If it is can be anything your code is wrong, as you should check if it is an array first before assuming :-)